tests/test_spstats.py

```diff
@@ -171,3 +170,49 @@ def test_patches_method(self):
             gsd=diff.res[0],
             area_size=10000
         )
+
+class TestBinning:
+
+    def test_nd_binning(self):
+
+        ref, diff, mask = load_ref_and_diff()
+
+        slope, aspect = xdem.coreg.calculate_slope_and_aspect(ref.data.squeeze())
```

Comment:
> I now had a thorough look and this looks super useful! Great job!!
> 
> I have a few comments, mostly not about your intended use, and I envision many to be able to use it for completely different purposes. Perhaps this could be made more generic to be a multi-purpose function (basically just rename the dh argument to something generic, modify the docstring a bit, and it's done!)?
> 
> Also, now that I've read the entire documentation and the tests, I more or less understand the functionality. I can unfortunately not say that I understand it completely, and I worry that this might be reflective of other users' experience as well. Could you please elaborate a bit on the docstrings, perhaps add a few minimal examples, and optimally also add documentation to readthedocs (this can of course be done in a later stage)?

Thanks a lot for the comments!! I was updating the function + parameters description with Amaury's comments when you reviewed, so hopefully it's already a lot better :)